Added extract descriptions from scanned images in a PDF

encamina-hramos opened the pull request:
- Various methods have been implemented in `ImageHelper` to extract scanned images from a PDF.
- An `ImageInfo` class has been added to encapsulate image types.
- The `ProcessImageWithVision` method has been modified to implement the new changes.

encamina-hramos committed:
Added extract descriptions from scanned images in a PDF

- Various methods have been implemented in `ImageHelper` to extract scanned images from a PDF.
- An `ImageInfo` class has been added to encapsulate image types.
- The `ProcessImageWithVision` method has been modified to implement the new changes.

lmarcos000 commented on src/Encamina.Enmarcha.SemanticKernel.Connectors.Document/SkVisionImageExtractor.cs:
```diff
@@ -80,34 +80,42 @@ public SkVisionImageExtractor(Kernel kernel, IOptions<SkVisionImageDocumentConne
     /// <returns>The extracted text description from the image.</returns>
     /// <exception cref="DocumentTooLargeException">Thrown when the image exceeds resolution limits or output capacity.</exception>
-    public string ProcessImageWithVision(Stream stream)
+    public string ProcessImageWithVision(Stream stream, int? rawCcittWidth = null, int? rawCcittHeight = null)
```
The comments for the two new parameters are missing 🙂

lmarcos000 commented on src/Encamina.Enmarcha.SemanticKernel.Connectors.Document/Utils/ImageHelper.cs:
```diff
@@ -15,85 +19,45 @@ public static class ImageHelper
-    /// <exception cref="ArgumentNullException">Thrown when the stream is null.</exception>
-    /// <exception cref="ArgumentException">Thrown when the stream does not contain a valid image.</exception>
-    public static (string MimeType, int Width, int Height) GetImageInfo(Stream stream)
+    public static (string MimeType, int Width, int Height) GetImageInfo(Stream stream, int? rawCcittWidth = null, int? rawCcittHeight = null)
```
This method is not being used

lmarcos000 replied:
And further down, GetMimeType and GetResolution don’t seem to be used either.

lmarcos000 commented on src/Encamina.Enmarcha.SemanticKernel.Connectors.Document/Utils/ImageHelper.cs:
```diff
@@ -185,4 +145,233 @@ public static (int Width, int Height) GetResolution(Stream stream)
+        {
+            line = reader.ReadLine();
+        }
+        while (line.StartsWith('#'));
```
This could be null

encamina-hramos committed:
Added comments, null validation and remove unused methods

- Handled null exception in the `ConvertPbmP4ToPng` loop.
- Removed unused methods:
  - `GetImageInfo`.
  - `GetMimeType`.
  - `GetResolution`.
- Added comments to the `ProcessImageWithVision` parameters.
